catalog/app/containers/NavBar/NavBar.tsx

```diff
@@ -141,16 +143,20 @@ export function Header({ children }: HeaderProps) {
   const classes = useHeaderStyles({
     customBg: !!settings?.theme?.palette?.primary?.main,
   })
+  const searchUIModel = SearchUIModel.useUnsafe()
   return (
     <M.Box>
       <M.Toolbar />
       <M.Slide appear={false} direction="down" in={!trigger}>
         <AppBar>
           <M.Toolbar disableGutters>
-            <M.Container className={classes.container} maxWidth="lg">
+            <LayoutContainer
+              className={classes.container}
+              fullWidth={searchUIModel?.state.view === SearchUIModel.View.Table}
```

Comment:
this seems like a crude ad-hoc workaround for the problem.
here's an example implementation of a value overrides from anywhere in the component tree: https://github.com/quiltdata/quilt/pull/4434

Reply:
Used `Provider` pattern to set `Layout.useFullWidth(true)` in Table view

Reply:
> Used Provider pattern to set Layout.useFullWidth(true) in Table view

that's a step forward, but the implementation is error-prone (especially in the case when several components try to override this state), because it relies on imperative state changes.